packages/helpermodules/update_config.py

```diff
@@ -1442,25 +1442,34 @@ def upgrade(topic: str, payload) -> None:
                 payload = decode_payload(payload)
                 if payload.get("type") == "sungrow" and "version" not in payload["configuration"]:
                     index = get_index(topic)
-                    # Assume an SG_WiNet version as default if no battery or counter found:
-                    version = 2
+                    version = None
+                    has_battery = False
                     for other_topic, other_payload in self.all_received_topics.items():
                         if re.search(f"openWB/system/device/{index}/component", other_topic) is not None:
                             child_component_payload = decode_payload(other_payload)
                             if child_component_payload.get("type") == "counter" and "version" in \
                                child_component_payload["configuration"]:
                                 version = child_component_payload['configuration']['version']
                                 log.debug(f"Version {version} found at counter for sungrow device {index}")
-                                # Pre-defined value found, stop component search
-                                break
                             elif child_component_payload.get("type") == "bat":
-                                # Assume a hybrid inverter since a battery is attached
-                                version = 3
-                    # Assume an SH_WiNet connection if SH_x has been set before, this is more compatible than SH_LAN:
-                    version = 3 if version == 0 else version
+                                has_battery = True
+
+                    if has_battery or version == 0:
+                        # Assume an SH_WiNet if hybrid inverter, this is compatible to SH_LAN (but not vice versa)
+                        version = 3
+                    elif not version:
+                        # Assume an SG_WiNet version as default if no battery or version from counter found
+                        version = 2
                     log.debug(f"Setting version {version} for sungrow device {index}")
                     payload["configuration"].update({"version": version})
-                Pub().pub(topic, payload)
+                    Pub().pub(topic, payload)
+
+                    pub_system_message(payload, "Die Konfiguration von Sungrow Geräten wurde aktualisiert. \
+                        Bitte die Version in den Geräteeinstellungen überprüfen", MessageType.WARNING)
+                    if has_battery:
+                        pub_system_message(payload, "Das Auslesen von Sungrow Hybrid-Wechselrichtern wurde \
+                            aktualisiert. Bitte den Speicher im Lastmanagement innerhalb des zugehörigen \
+                            Hybrid-Wechselrichters anordnen", MessageType.WARNING)
```

Comment:
Das sollte automatisiert erfolgen. Das führt sonst zu Rückfragen bei den Kunden und sie sind genervt, wenn Sie bei einem Update händisch die Einstellungen anpassen müssen.
Diese Schleife `for other_topic, other_payload in self.all_received_topics.items():` kannst Du nutzen, um die IDs von WR und Speicher mit `get_second_index` herauszubekommen und `openWB/counter/get/hierarchy`in einer Variablen abzulegen. Dann kannst Du die Methoden zur Bearbeitung der Hierarchie aus CounterAll benutzen:
```
counter_all = CounterAll()
counter_all.data.get.hierarchy = hierarchy
counter_all.hierarchy_remove_item(bat_index)
counter_all.hierarchy_add_item_below(bat_index, ComponentType.BAT, inverter_index)
Pub().pub("openWB/counter/get/hierarchy", counter_all.data.get.hierarchy)
```
(ungetestet)

Reply:
Danke! Habe ich umgesetzt und getestet für gemsichte SH/SG/Bat Setups